### ISSUE TITLE
Offline with non persistent licenses

### ISSUE THREAD

**Issue by sarge** [google/shaka-player#1159]
Related but I think distinct to #512 

I am using Offline with non persistent licenses, on Chrome (63) which triggers an error ```NO_INIT_DATA_FOR_OFFLINE``` due to the sessions list being empty.

https://github.com/google/shaka-player/blob/master/lib/offline/storage.js#L754

My use case is downloading the file for online playback. I have not attempted to play the content, instead are downloading the content before any license request is made. 
The initdata is stored in the file / not the manifest.

My fix has been to only use the sessions if  ```{usePersistentLicenses:true}```. Is that the intent of this check? 
```
if (!sessions.length) {
```
to 
```
if (!sessions.length && this.config_.usePersistentLicense) {
 ```

Cheers
Mark

**Comment by joeyparrish** [comment]
That definitely sounds like a bug, and your suggested fix makes sense.  We will try to reproduce the issue and then try your suggestion.  Thanks!

**Comment by joeyparrish** [comment]
Confirmed the bug.  It turns out that we already have a test for this, but it was being skipped because of a typo introduced in 892d1074336283c4f05c8b10b1555164345ee7a9.  So we lost test coverage about 6 months ago.  Still tracking down when we broke the feature itself.

**Comment by joeyparrish** [comment]
It looks like the temporary licensing feature wasn't broken per se, but the check for init data with temporary licenses was always broken.

If there is no init data in the manifest, the only way we can create a session and request a license is by playing the content.  This is why init data is required for offline storage (with persistent licenses).  We have to get a persistent license without playing the content, so the init data must be available upfront.

This is not true for storage with temporary licenses.  In that scenario, there is no need for init data, because there is no need for a license at all.

In our existing temporary license storage test, we store content that has init data upfront, so we never triggered that error in testing.

**Comment by joeyparrish** [comment]
The test that was accidentally disabled for 6 months is somewhat orthogonal to this issue.  Re-enabling the test triggered a test failure, but it was an unrelated test failure caused by flaky timing in the test itself.

**Comment by sarge** [comment]
Thanks for the update Joey. I'm impressed by the rigour of you and your team:)

**Comment by joeyparrish** [comment]
Thanks!  That's kind of you to say.

**Comment by joeyparrish** [comment]
The fix has been cherry-picked for v2.2.9.
